cmd/influxd/server_integration_test.go

```diff
@@ -654,3 +658,101 @@ func Test3NodeServer(t *testing.T) {
 
 	runTestsData(t, testName, nodes, "mydb", "myrp")
 }
+
+func TestClientLibrary(t *testing.T) {
+	testName := "single server integration via client library"
+	if testing.Short() {
+		t.Skip(fmt.Sprintf("skipping '%s'", testName))
+	}
+	dir := tempfile()
+	defer func() {
+		os.RemoveAll(dir)
+	}()
+
+	database := "mydb"
+	retentionPolicy := "myrp"
+	now := time.Now().UTC()
+
+	nodes := createCombinedNodeCluster(t, testName, dir, 1, 8290)
+	createDatabase(t, testName, nodes, database)
+	createRetentionPolicy(t, testName, nodes, database, retentionPolicy)
+
+	tests := []struct {
+		name                         string
+		bp                           client.BatchPoints
+		results                      client.Results
+		query                        client.Query
+		writeExpected, queryExpected *client.Results
+		writeErr, queryErr           string
+	}{
+		{
+			name:          "empty batchpoint",
+			writeExpected: &client.Results{Err: fmt.Errorf("database is required")},
```

Comment:
All these tests need to move away from the types, and just do string compares -- using types in the integration tests is too brittle, and too difficult to read.

Can you check out the table-driven tests in the other parts of this file to see what I mean?